pypeit/edgetrace.py

```diff
@@ -5037,6 +5075,49 @@ def order_refine(self, debug=False):
         if self.spectrograph.pypeline != 'Echelle':
             msgs.warn('Parameter add_missed_orders only valid for Echelle spectrographs.')
             return
+        
+        if not self.can_pca():
+            msgs.error('Refining the orders currently requires a PCA decomposition of the '
+                       'order edges.')
+
+        # Update the PCA
+        self.build_pca()
+
+        if self.spectrograph.ech_fixed_format:
+            # TODO: Shouldn't this also need reference_row?
```

Comment:
still a valid TODO?

Reply:
Yes, but I've moved and expanded on it.  We need to test adding a missing order that's found by the hard-coded expectations for where the orders are (in a different PR).